### ISSUE TITLE
pod install limbo

### ISSUE THREAD

**Issue by gdesoto** [microsoft/appcenter-sdk-cordova#118]
### **Description**

I'm trying to get the Cordova working in Azure DevOps hosted pipeline. Theoretically, this would create a clean environment for building my cordova project. It used to fail when trying to install the iOS platform because the AppCenter cordova sdk would run a pod install. I found that solution to be to run "pod repo update" prior to building cordova. That was working fine for a while...

However, it looks like they've released cocoapods 1.8.x, and running "pod repo update" beforehand no longer is an option because [this reported cocoapods issue](https://github.com/CocoaPods/CocoaPods/issues/9226) where it no longer creates a repo directory on update.

Their solution is to run "pod install --repo-update". However, I cannot do this since the podfile doesn't exist prior to installing the platform and plugin. So now my builds are stuck. I can't run pod repo update and I can't run pod install as both fail. What is the suggested method for getting the sdk to successfully install?

### **Repro Steps**

Please list the steps used to reproduce your issue.

1. Start with a clean environment (no platform folders in Cordova), and add the ios platform and appcenter plugins to the cordova config.xml file.
2. Run cordova platform add ios
3. If you haven't run pod repo update, you get the following error messages
`Failed to restore plugin "cordova-plugin-appcenter-analytics" from config.xml. You might need to try adding it again. Error: The CocoaPods repo has not been synced yet, this will take a long time (approximately 500MB as of Sept 2016). Please run `pod setup` first to sync the repo.`
4. If you run cordova build ios, you get the following error:
`fatal error: module 'AppCenter' not found`

### **Details**

1. Which SDK version are you using?
    - 0.4.1
2. Which OS version did you experience the issue on?
    - iOS 
3. What version of Cordova, Cordova platforms, Cocoapods, and Ruby are you using?
    - Cordova version 9.0.0
    - cordova-ios@^5.0.1
    - Cocoapods 1.8.1 & 1.8.3
4. What device version did you see this error on?  Were you using a simulator or a physical device?
    - Unable to get a successful platform install or build.
5. What third party plugins are you using?
    - cordova-plugin-appcenter-analytics 0.4.1 "App Center Analytics for Cordova"
    - cordova-plugin-appcenter-crashes 0.4.1 "App Center Crashes for Cordova"
    - cordova-plugin-appcenter-shared 0.4.1 "App Center shared code for Cordova"
    - cordova-plugin-ionic-webview 4.1.2 "cordova-plugin-ionic-webview"

**Comment by adriano-kaiser** [comment]
I am having exactly the same issue. The Azure Devops macOS 10.14 pre installed cocoapod has been updated. See:
https://github.com/microsoft/azure-pipelines-image-generation/commit/68f04f203af798a392e8c331f74d2b9b6273176e

I did tried to downgrade to 1.7.5. It also did not worked. So far i only managed to make it work, but with some very very ugly workarounds. I removed the ios platform and also the affected plugins (Push Plugin has the same issue) from package.json. Then here my awful workaround:

npm -g i ionic cordova   
ionic cordova plugin add cordova-plugin-appcenter-analytics
ionic cordova plugin add cordova-plugin-appcenter-crashes
ionic cordova plugin add phonegap-plugin-push --variable FCM_VERSION="11.0.1"
ionic cordova platform add ios
pod install --project-directory=platforms/ios  
pod repo update
pod install --project-directory=platforms/ios
rm -rf platforms
rm -rf plugins
ionic cordova platform add ios
pod install --project-directory=platforms/ios
ionic cordova build ios --prod --release --device --buildConfig=build-enterprise.json -- --buildFlag=-UseModernBuildSystem=0

**Comment by gdesoto** [comment]
I was previously using the hooks "before_platform_add" for the pod repo update and "after_platform_add" for the pod install project dir command.

I'm currently trying to get your workaround steps added. I'm using Vue / Quasar instead of Ionic, but we should reach the same outcome. I'll update with any success.

**Comment by dexter31** [comment]
Had the same problem. Was running a CI on Azure Devops. Adding a pre build script to downgrade resolved the issue, with an annoying 6 minute trade off.

`sudo gem uninstall cocoapods

sudo gem install cocoapods -v 1.7.5

pod setup`

**Comment by adriano-kaiser** [comment]
@dexter31 that one works for me also! I also tried to downgrade but it was not working. Basically i had `pod repo update` instead of `pod setup`. Thanks for the insights!

**Comment by kommundsen** [comment]
> Had the same problem. Was running a CI on Azure Devops. Adding a pre build script to downgrade resolved the issue, with an annoying 6 minute trade off.
> 
> `sudo gem uninstall cocoapods`
> 
> `sudo gem install cocoapods -v 1.7.5`
> 
> `pod setup`

Thank you for saving the day. Celebrating at Burger King!

**Comment by Jamminroot** [comment]
Yeah, that seem to be a problem with cocoapods. While this being fixed, can you check, whether [suggested trick](https://github.com/microsoft/appcenter-sdk-cordova/issues/118/#issuecomment-540397102) works for you, @gdesoto?

**Comment by gdesoto** [comment]
That did it! And without build warnings! I'm also not a fan of the additional time to build, but I'll take what I can get for the moment. Thanks @dexter31  and @adriano-kaiser.

**Comment by Jamminroot** [comment]
Glad the problem is resolved!
Feel free to reopen the ticket if you face the issue again.

**Comment by vbraun** [comment]
Why is this issue closed? Installation is broken on current OSX, and the only workaround is to downgrade cocoapods. Surely this is not the intended mode of operation. I didn't see a different ticket tracking this issue either.

**Comment by kjng** [comment]
@vbraun, I think in order to get this problem resolved, we should add conversation to this issue I created in the cordova-ios repository: https://github.com/apache/cordova-ios/issues/700

That project is the one that can resolve this issue (by supporting latest cocoapods).

**Comment by mibzman** [comment]
If anyone is still having this issue like I was, it can be caused by a single quote in the project name.  My workaround and the relevant issue is here: [apache/cordova-ios#717](https://github.com/apache/cordova-ios/issues/717#issuecomment-567161530)
